File: Data Structures/Java/Queue.java
import java.util.LinkedList;

class Queue {
    private LinkedList<Integer> list;

    public Queue() {
        list = new LinkedList<>();
    }

    public boolean isEmpty() {
        return list.isEmpty();
    }

    public void enqueue(int item) {
        list.addLast(item);
    }

    public int dequeue() {
        if (!isEmpty()) {
            return list.removeFirst();
        } else {
            throw new IllegalStateException("Queue is empty");
        }
    }

    public int front() {
        if (!isEmpty()) {
            return list.getFirst();
        } else {
            throw new IllegalStateException("Queue is empty");
        }
    }

    public static void main(String[] args) {
        Queue queue = new Queue();
        queue.enqueue(10);
        queue.enqueue(20);
        queue.enqueue(30);

        System.out.println("Front element: " + queue.front());

        System.out.println("Dequeued element: " + queue.dequeue());
        System.out.println("Dequeued element: " + queue.dequeue());

        System.out.println("Front element after dequeues: " + queue.front());
    }
}

File: Data Structures/C/Queue.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

struct Queue {
    int arr[MAX_SIZE];
    int front, rear;
};

typedef struct Queue Queue;

void init(Queue* queue) {
    queue->front = -1;
    queue->rear = -1;
}

int is_empty(Queue* queue) {
    return queue->front == -1 && queue->rear == -1;
}

int is_full(Queue* queue) {
    return (queue->rear + 1) % MAX_SIZE == queue->front;
}

void enqueue(Queue* queue, int item) {
    if (!is_full(queue)) {
        if (is_empty(queue)) {
            queue->front = 0;
        }
        queue->rear = (queue->rear + 1) % MAX_SIZE;
        queue->arr[queue->rear] = item;
    } else {
        printf("Queue is full\n");
    }
}

int dequeue(Queue* queue) {
    int item = -1;
    if (!is_empty(queue)) {
        item = queue->arr[queue->front];
        if (queue->front == queue->rear) {
            init(queue);
        } else {
            queue->front = (queue->front + 1) % MAX_SIZE;
        }
    } else {
        printf("Queue is empty\n");
    }
    return item;
}

int front(Queue* queue) {
    if (!is_empty(queue)) {
        return queue->arr[queue->front];
    } else {
        printf("Queue is empty\n");
        return -1;
    }
}

int main() {
    Queue queue;
    init(&queue);

    enqueue(&queue, 10);
    enqueue(&queue, 20);
    enqueue(&queue, 30);

    printf("Front element: %d\n", front(&queue));

    printf("Dequeued element: %d\n", dequeue(&queue));
    printf("Dequeued element: %d\n", dequeue(&queue));

    printf("Front element after dequeues: %d\n", front(&queue));

    return 0;
}
